starter-code/public/javascripts/strength.js

```diff
@@ -0,0 +1,162 @@
+            var specialchars = new RegExp('([!,%,&,@,#,$,^,*,?,_,~])');
+
+            function GetPercentage(a, b) {
+                    return ((b / a) * 100);
+                }
+
+                function check_strength(thisval,thisid){
+                    if (thisval.length > 8) { characters = 1; } else { characters = -1; };
+                    if (thisval.match(upperCase)) { capitalletters = 1} else { capitalletters = 0; };
+                    if (thisval.match(lowerCase)) { loweletters = 1}  else { loweletters = 0; };
+                    if (thisval.match(numbers)) { number = 1}  else { number = 0; };
+
+                    var total = characters + capitalletters + loweletters + number + special;
+                    var totalpercent = GetPercentage(7, total).toFixed(0);
+
+                    if (!thisval.length) {total = -1;}
+
+                    get_total(total,thisid);
+                }
+
+            function get_total(total,thisid){
+
+                var thismeter = $('div[data-meter="'+thisid+'"]');
+                    if (total <= 1) {
+                   thismeter.removeClass();
+                   thismeter.addClass('veryweak').html('very weak');
+                } else if (total == 2){
+                    thismeter.removeClass();
+                   thismeter.addClass('weak').html('weak');
+                } else if(total == 3){
+                    thismeter.removeClass();
+                   thismeter.addClass('medium').html('medium');
+
+                } else {
+                     thismeter.removeClass();
+                   thismeter.addClass('strong').html('strong');
+                }
+                
+                if (total == -1) { thismeter.removeClass().html('Strength'); }
+            }
+
+
+
+
+
+            var isShown = false;
+            var strengthButtonText = this.options.strengthButtonText;
+            var strengthButtonTextToggle = this.options.strengthButtonTextToggle;
+
+
+            thisid = this.$elem.attr('id');
+
+            this.$elem.addClass(this.options.strengthClass).attr('data-password',thisid).after('<input style="display:none" class="'+this.options.strengthClass+'" data-password="'+thisid+'" type="text" name="" value=""><a data-password-button="'+thisid+'" href="" class="'+this.options.strengthButtonClass+'">'+this.options.strengthButtonText+'</a><div class="'+this.options.strengthMeterClass+'"><div data-meter="'+thisid+'">Strength</div></div>');
+             
+            this.$elem.bind('keyup keydown', function(event) {
+                thisval = $('#'+thisid).val();
+                $('input[type="text"][data-password="'+thisid+'"]').val(thisval);
+                check_strength(thisval,thisid);
+                
+            });
+
+             $('input[type="text"][data-password="'+thisid+'"]').bind('keyup keydown', function(event) {
+                thisval = $('input[type="text"][data-password="'+thisid+'"]').val();
+                console.log(thisval);
```

Comment:
Be careful with the libraries that you put in your project... this console.log print your password in the console...🐛